commons/hu.qgears.opengl.commons/kms/drmScreenshot.cpp

```diff
@@ -101,67 +101,66 @@ int main (int argc, char ** argv)
 	}
 
 	/* find the first available connector with modes */
-	for(i=0; i < resources->count_connectors; ++i){
+	for(i=0; i < resources->count_connectors; ++i) {
 		connector = drmModeGetConnector(fd, resources->connectors[i]);
 		if(connector != NULL){
-			fprintf(stderr, "connector %d found\n", connector->connector_id);
-			if(connector->connection == DRM_MODE_CONNECTED
-				&& connector->count_modes > 0)
-				break;
-			drmModeFreeConnector(connector);
-		}
-		else
+			fprintf(stderr, "connector %d found; mode count: %d; state: ", 
+			    connector->connector_id, connector->count_modes);
+			    
+			if(connector->connection == DRM_MODE_CONNECTED) {
+				fprintf(stderr, "connected\n");
+				
+				if (first_good_connector == NULL) {
+				    first_good_connector = connector;
+				}
+			} else {
+			    fprintf(stderr, "disconnected\n");
+			}
+//				break;
+//			drmModeFreeConnector(connector);
+		} else {
 			fprintf(stderr, "get a null connector pointer\n");
+		}
 	}
-	if(i == resources->count_connectors){
+	
+	connector = first_good_connector;
+	
+	if (first_good_connector == NULL) {
 		fprintf(stderr, "No active connector found.\n");
 		goto free_drm_res;
 	}
 
 	mode = connector->modes[0];
-	fprintf(stderr, "(%dx%d)\n", mode.hdisplay, mode.vdisplay);
+	fprintf(stderr, "connector mode 0: %dx%d\n", mode.hdisplay, mode.vdisplay);
```

Comment:
perror, fprintf, printf : ezek helyett egy log metódos volna jó szerintem, hogy könnyen tudjuk ki / be kapcsolni a logüzeneteket.

Reply:
így van, ha egyértelmű lesz, hogy behatóbban kell foglalkozni a kmsgl backenddel